napari/layers/utils/layer_utils.py

```diff
@@ -278,7 +300,312 @@ def calc_data_range(data, rgb=False) -> Tuple[float, float]:
+            ),
+            slice(-slice_size, None),
+        ]
+
+    chunk_size_product = np.prod(chunk_size)
+    allowed_chunks = int(PIXEL_THRESHOLD // chunk_size_product)
+    multiplier = min(allowed_chunks // n_slices, MAX_NUMBER_OF_CHUNKS)
+
+    if chunk_size_product > PIXEL_THRESHOLD:
+        return None
+    if shape[0] >= 3:
+        if multiplier >= 1:
+            slices = [
+                slice(0, chunk_size[0] * multiplier),
+                slice(
+                    center - chunk_size[0] * (multiplier // 2 - 1),
+                    center + chunk_size[0] * multiplier // 2,
+                ),
+                slice(-chunk_size[0] * multiplier, -1),
+            ]
+        else:
+            # Means we have less than 3 allowed chunks so we just take a center slice.
+            slices = [slice(center, center + chunk_size[0])]
+    # due to earlier check we now data is above pixel threshold so 2 chunks is above as well
+    else:  # shape[0] == 2 or 1
+        slices = [slice(0, chunk_size[0])]
+
+    return slices
+
+
+def _get_blocks_grid_shape(
+    data_shape: Sequence[int], chunk_size: Sequence[int]
+) -> Tuple[int, ...]:
+    """
+    Get the approximate shape of the grid of array chunks.
+
+    Gets the shape of the array not on the pixel level but on the chunk level. For example if x is 100 and
+    chunk size is 10, 10 chunks fit in 100 and will thus be returned. In case of x being 105, still 10 will be
+    returned. This to ensure that only whole chunks will be loaded.
+
+    Parameters
+    ----------
+    data_shape: tuple[int, ...]
+        The shape of an array of chunked data.
+    chunk_size: tuple[int, ...]
+        The size per dimension of the chunks in the chunked data array.
+
+    Returns
+    -------
+    tuple[int, ...]
+        shape indicating the number of chunks per dimension.
+    """
+    return tuple(
+        data_shape[i] // chunk_size[i] for i in range(len(data_shape))
+    )
+
+
+def _get_plane_indices(
+    shape: Sequence[int], offset: int
+) -> List[Tuple[int, ...]]:
+    """
+    Get the indices that correspond to the lowest, middle and highest index of the non-visible dimensions in shape.
+
+    In case the different planes are the same, duplicates are removed.
+
+    Parameters
+    ----------
+    shape: Sequence[int]
+        Either the shape of the raw data (in pixels) or the block shape of the data (chunks)
+    offset: int
+        Number of visible dimensions
+
+    Returns
+    -------
+    idxs: list[tuple[int, ...]]
+        Bottom, middle and top plane for each non-visible dimension or single plane
+    """
+    bottom_plane_idx = (0,) * (len(shape) - offset)
+    middle_plane_idx = tuple(s // 2 for s in shape[:-offset])
+    top_plane_idx = tuple(s - 1 for s in shape[:-offset])
+    idxs = [bottom_plane_idx, middle_plane_idx, top_plane_idx]
+    if len(idxs) != len(set(idxs)):
+        return list(set(idxs))
+    return idxs
```

Comment:
```suggestion
    return list(set(idxs))
```
If we need to create a set, then just use it.